web-registry/src/pages/Dashboard/MyEcocredits/hooks/useBasketTakeSubmit.tsx

```diff
@@ -58,7 +73,33 @@ const useBasketTakeSubmit = ({
         memo: values?.retirementNote,
       };
 
-      await signAndBroadcast(tx, () => setBasketTakeTokens(undefined));
+      const onError = (err?: Error): void => {
+        track<'takeFromBasketFailure', TakeFromBasketFailure>(
+          'takeFromBasketFailure',
+          {
+            basketName: basket?.name,
+            quantity: amount,
+            retireOnTake: values.retireOnTake,
+            errorMessage: err?.message,
+          },
+        );
+      };
+      const onSuccess = (deliverTxResponse?: DeliverTxResponse): void => {
+        const batchesFromTake = takeEventToBatches(deliverTxResponse!);
```

Comment:
this is how i chose to include information in the tracking data about which batch denom's the credits came from during a take action. this function came from the code that was being used to populate this same exact info in the success modal.

i was unsure of a clean way to share this data between the success modal and the onSuccess shown here. i tried it a few different ways by passing props and attempting to use a `useCallback`.

i could not find a solution to this problem without the duplicated calculation. my approach is to re-compute the batch denoms for the tx response. i'm open to a better way of doing this but maybe we can handle it separately, since this works and is a relatively small computational overhead.

Reply:
I set up a global performant context (that prevents rerendering) to share variables between deeply nested components in a previous PR.
You can find a usage here:
Set: https://github.com/regen-network/regen-web/blob/dev/web-registry/src/hooks/useMsgClient.ts#L65
Get: https://github.com/regen-network/regen-web/blob/dev/web-registry/src/components/organisms/WalletButton/WalletButton.tsx#L25
This might be helpful.

Reply:
i'm going to keep this in mind going forward but choosing not to implement this right now